torchinductor/codegen/cpp.py

```diff
@@ -97,43 +99,12 @@ def argmax_argmin_prefix(reduction_type, src_dtype, tmpvar):
 
 @functools.lru_cache()
 def cpp_prefix():
-    _, filename = codecache.write(
-        textwrap.dedent(
-            """
-            #include <algorithm>
-            #include <atomic>
-            #include <cmath>
-            #include <cstdlib>
-            #include <limits>
-            #include <omp.h>
-
-            #include <ATen/core/PhiloxRNGEngine.h>
-
-            template<typename T>
-            inline T mod(T a, T b) { return a % b; }
-            template<>
-            inline float mod(float a, float b) { return std::fmod(a, b); }
-            template<>
-            inline double mod(double a, double b) { return std::fmod(a, b); }
-
-            constexpr float uint32_to_uniform_float(uint32_t value) {
-                // maximum value such that `MAX_INT * scale < 1.0` (with float rounding)
-                constexpr float scale = 4.6566127342e-10;
-                return static_cast<float>(value & 0x7FFFFFFF) * scale;
-            }
-
-            float normalized_rand_cpu(uint32_t seed, uint32_t offset) {
-                return uint32_to_uniform_float(at::Philox4_32(seed, 0, offset)());
-            }
-
-            float randn_cpu(uint32_t seed, uint32_t offset) {
-                at::Philox4_32 engine(seed, 0, offset);
-                return engine.randn(10);
-            }
-            """
-        ),
-        "h",
-    )
+    path = Path(__file__).parent / "cpp_prefix.h"
+    with path.open() as f:
+        _, filename = codecache.write(
```

Comment:
Add a comment saying we make a copy so caches get invalidated properly

Reply:
Sorry, can you explain this to me? who is the comment for?

Reply:
Ah perhaps you got this right by mistake.  This code basically just copies `cpp_prefix.h` to our cache directory.

Someone reading this might think they could just `#include ".../cpp_prefix.h"` directly.  This would be incorrect because if you modify `cpp_prefix.h` it won't invalidate the code cache.

Reply:
> Ah perhaps you got this right by mistake. 

The only way I get things right. 

> This code basically just copies cpp_prefix.h to our cache directory.

The old code wrote the contents of a cpp file, from a string. The new code write the contents of a cpp file, from a file.  

I am not sure what you want me to change here.

Reply:
This code is correct, don't worry about it.  I was just suggesting a comment explaining why we need it this way.